domain/application/state/application.go

```diff
@@ -2193,6 +2194,412 @@ FROM v_revision_updater_application_unit
 	})
 }
 
+// GetApplicationConfig returns the application config attributes for the
+// configuration.
+// If no application is found, an error satisfying
+// [applicationerrors.ApplicationNotFound] is returned.
+func (st *State) GetApplicationConfig(ctx context.Context, appID coreapplication.ID) (map[string]application.ApplicationConfig, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, internalerrors.Capture(err)
+	}
+
+	// We don't currently check for life in the old code, it might though be
+	// worth checking if the application is not dead.
+	ident := applicationID{ID: appID}
+
+	// This isn't ideal, as we could request this in one query, but we need to
+	// perform multiple queries to get the data. First is to get the application
+	// availability, second to just get the application overlay config for the
+	// charm config and the application settings for the trust config.
+	appQuery := `
+SELECT &applicationID.*
+FROM application
+WHERE uuid = $applicationID.uuid;
+`
+	configQuery := `
+SELECT &applicationConfig.*
+FROM v_application_config
+WHERE uuid = $applicationID.uuid;
+`
+	settingsQuery := `
+SELECT &applicationSettings.*
+FROM application_setting
+WHERE application_uuid = $applicationID.uuid;`
+
+	appStmt, err := st.Prepare(appQuery, ident)
+	if err != nil {
+		return nil, internalerrors.Errorf("preparing query for application config: %w", err)
+	}
+	configStmt, err := st.Prepare(configQuery, applicationConfig{}, ident)
+	if err != nil {
+		return nil, internalerrors.Errorf("preparing query for application config: %w", err)
+	}
+	settingsStmt, err := st.Prepare(settingsQuery, applicationSettings{}, ident)
+	if err != nil {
+		return nil, internalerrors.Errorf("preparing query for application config: %w", err)
+	}
```

Comment:
knit: All these error messages are identical. They should refer to the particular statement that has failed to prepare